server/routes/matching/match_jobs.py

```diff
@@ -46,94 +46,118 @@ def match_jobs():
+
+                    if len(candidate.json()["candidates"]["soft_skills"]) > 0:
+                        candidate_soft_skills = candidate.json()["candidates"][
+                            "soft_skills"
+                        ]
+                        cand_soft_skills.append(candidate_soft_skills)
 
                 jobs_response = requests.get(f"{domain_name}/api/get_all_jobs")
-                print("JOBS", jobs_response.json())
                 if jobs_response.status_code == 200:
                     jobs = jobs_response.json()["jobs"]
 
                     for job in jobs:
-                        job_skills = [skill["skill_name"] for skill in job["skills"]]
-                       
-                        job_id = job["id"]
-
-                        count = 4
-                        total_score = 0
-
-                        if job_skills:
-                            job_tech_score = score(job_skills, cand_skills, cand_levels)
-                            total_score += 4 * job_tech_score
-
-                            if job["values"]:
-                                count += 1
-                                if cand_values:
-                                    job_val_score = score(job["values"], cand_values)
-                                    total_score += job_val_score
-
-                            if job["soft_skills"]:
-                                count += 2
-                                if cand_soft_skills:
-                                    job_soft_score = score(
-                                        job["soft_skills"], cand_soft_skills
-                                    )
-                                    total_score += 2 * job_soft_score
+                        if len(job["skills"]) > 0:
+                            job_skills = [
+                                skill["skill_name"] for skill in job["skills"]
+                            ]
 
-                            job_score = round(total_score / count, 1)
+                            job_id = job["id"]
+
+                            count = 4
+                            total_score = 0
+
+                            if job_skills and len(cand_skills) > 0:
+                                job_tech_score = score(
+                                    job_skills, cand_skills, cand_levels
+                                )
+                                total_score += 4 * job_tech_score
 
-                            if job_score >= 60:
-                                job_match.append({"id": job_id, "score": job_score})
-                                if job["matching_candidates"]:
-                                    duplicate = [
-                                        ix
-                                        for ix, cand in enumerate(
-                                            job["matching_candidates"]
+                                if job["values"] and len(cand_values) > 0:
```

Comment:
```suggestion
                                if job["values"]:
```

Reply:
here if you don't check the len of the canidates, in the score functions, you'll try to loop for `candidate_skills = [
        "_".join(skill.lower().split(" ")) for skill in candidate_skills
    ]` which fails the call because skills can be either **None** or **empty** `[]`